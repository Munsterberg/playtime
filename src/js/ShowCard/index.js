import React, { PropTypes } from 'react';

const propTypes = {
    show: PropTypes.object.isRequired
};

function ShowCard({ show }) {
    return (
        <div className="show-card">
            <img src={`/public/img/posters/${show.poster}`} alt={show.title} />
            <div>
                <h3>{show.title}</h3>
                <h4>({show.year})</h4>
                <p>{show.description}</p>
            </div>
        </div>
    );
}

ShowCard.propTypes = propTypes;

export default ShowCard;
