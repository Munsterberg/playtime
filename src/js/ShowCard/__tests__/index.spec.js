import React from 'react';

import ShowCard from '../index';

describe('<ShowCard />', () => {
    let wrapper = null;
    const props = {
        show: {
            description: 'A movie description for testing',
            imdbID: 'tt1251',
            poster: 'something.jpg',
            title: 'Test movie',
            trailer: 'DfaaomAN',
            year: '2015'
        }
    };

    beforeEach(() => {
        wrapper = shallow(<ShowCard {...props}/>);
    });

    it('should render correct component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
