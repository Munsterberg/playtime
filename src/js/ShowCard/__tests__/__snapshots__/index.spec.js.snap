// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowCard /> should render correct component 1`] = `
<div
  className="show-card"
>
  <img
    alt="Test movie"
    src="/public/img/posters/something.jpg"
  />
  <div>
    <h3>
      Test movie
    </h3>
    <h4>
      (
      2015
      )
    </h4>
    <p>
      A movie description for testing
    </p>
  </div>
</div>
`;
